### DISCUSSION TITLE
How can I fix this issue in realtime api?

### DISCUSSION THREAD

**Discussion by rajesh6161** [pocketbase/pocketbase#795] [Q&A]
{
  "errorDetails": "No client associated with connection ID \"QN09nMOF2C1Dh0DGfQCoGY67jKea1CAAOESzAchk\"",
  "errorMessage": "Missing or invalid client id."
}
I am using pockethost.io and this is what I am getting in logs.
This was working fine in localhost.

**Comment by rajesh6161** [comment]
I guess creating a new instance of Pocketbase causes this issue

**Reply by ganigeorgiev** [reply]
I'm not sure. There is not enough information to debug the issue. If you can't reproduce it locally and think that it is related to pockethost, it's better to open an issue in the related benallfree's repo - https://github.com/benallfree/pockethost.

**Comment by rajesh6161** [comment]
Hey @ganigeorgiev I am getting this issue everytime `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://abiding-sundown.pockethost.io/api/realtime. (Reason: CORS request did not succeed). Status code: (null).`
And I am using  @benallfree 's `pockethost`. How can I fix this issue? plus suggest me how to properly setup realtime If I am wrong.

And the real time is working totally fine in localhost.

**Comment by benallfree** [comment]
@rajesh6161 Just tagging this with https://github.com/benallfree/pockethost/issues/10 where you noted this issue.

I'd be interested to hear from @ganigeorgiev as to how exiting/cycling the `pocketbase` process would affect realtime connections. If it will cause realtime connections to be dropped or events to be missed, perhaps we could discuss how best to detect active SSE connections.

**Reply by rajesh6161** [reply]
It also gives me this issue while using pockehost.io: [apis.js:13](e:/realtime/src/utils/apis.js)
Uncaught (in promise) ClientResponseError 404: Missing or invalid client id.

**Reply by ganigeorgiev** [reply]
@benallfree All realtime connections are dropped If the PocketBase process is terminated.

> how best to detect active SSE connections

You can check if there are any connected clients by inspecting `len(app.SubscriptionsBroker().Clients())`.

**Reply by benallfree** [reply]
I was wondering more about monitoring proxy connections and shutting down the PocketBase instance after a period of connection inactivity at the proxy level. I remember you saying that PocketBase keeps the connection info for a period of time in case the client reconnects, so maybe I should mirror that behavior. I assume you mean that you'll buffer realtime events and send them out if the client reconnects within a period of time?

 I could use `len(app.SubscriptionsBroker().Clients())` if there were a way to query it. I could make a custom API route, but another cool idea (maybe?) would be to make an admin-only realtime 'instance status' endpoint that would send realtime updates, something like this:
 
 ```ts
const client = new PocketBase(...)
client.realtime.subscribe(`__pocketbase__`, e=>{
   console.log(e.record)
});
```

```
// This is a __pocketbase__ structure, containing whatever kind of metadata PocketBase wants to publish about itself
{
   realtimeConnectionCount: 42,
   uptime: 882832
   dbSize: 183882
}
```

**Reply by ganigeorgiev** [reply]
@benallfree 

> I assume you mean that you'll buffer realtime events and send them out if the client reconnects within a period of time?

No, currently there is no buffering (although there are plans to add support for that in the future). The server has 5min idle timeout, meaning that if the client (eg. browser) doesn't receive any new messages for that period, the connection will be dropped. When that happen, if the client is still active and wants to listen to the realtime changes, it will send a reconnect request. And the same process will repeat until there are no subscriptions or one of the sides is no longer available.

**Reply by benallfree** [reply]
Ok so if the client temporarily drops its connection, PocketBase will not send a backlog of realtime updates upon reconnect.

Thanks, I will look into this more. I am hoping I can still shut down PocketBase during idle time at least for the free hosting tier, but I don't want to adversely affect realtime behavior.
